content/learning-paths/cross-platform/integer-vs-floats/type-demotion-problems.md

```diff
@@ -78,12 +78,12 @@ a = -16380, b = -16380000, c = 799869184, d1 = 4608, d2 = 4608
 e1 = 2147483647, e2 = 2147483647
 ```
 
-Obviously `w` is a huge value and it does not fit in a `float` (remember largest positive float is `3.4e+38`), but the compiler only complains about `z` which uses bracket initialization and not `y` which uses assignment. 
+Obviously `w` is a huge value and it does not fit in a `float` (remember that the largest positive float is `3.4e+38`), but the compiler only complains about `z` which uses bracket initialization and not `y` which uses assignment. 
 
-Similarly, `d1` an `int16_t` uses assignment and does not generate a warning, but `d2`, also an `int16_t`, uses bracket initialization and trigger a compiler warning.
+Similarly, `d1` and `int16_t` use assignment and do not generate a warning but `d2`, also an `int16_t`, use bracket initialization and trigger a compiler warning.
```

Comment:
this actually denotes that `d1` is of type `int16_t`, not that there are two entities.

Reply:
reverted back